### PR TITLE
adding clang-format for cpp files

### PR DESCRIPTION
**Pull request by hyperbolic2346**
Pulling over the clang-format rules from cudf. Unsure the feelings on clang-format or pre-commit, but it seemed reasonable to follow cudf's lead here.

Signed-off-by: Mike Wilson <knobby@burntsheep.com>

### CODE REVIEW

**Comment by hyperbolic2346** [comment]
build

File: src/main/cpp/.clang-format [hunk 1]
```diff
@@ -0,0 +1,155 @@
+---
```
Comment (gerashegalov) [inline src/main/cpp/.clang-format:1]: it looks like this is merely an exact copy of  cudf's `.clang-format`. We should use the original file to stay in sync if possible

Can you try adding 
```
-style='file:thirdparty/cudf/cpp/.clang-format' 
```
and remove the copy if it works?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+repos:
+      - repo: https://github.com/pre-commit/mirrors-clang-format
+        rev: v11.1.0
+        hooks:
+              - id: clang-format
+                files: \.(cu|cuh|h|hpp|cpp|inl)$
+                types_or: [file]
+                args: ['-fallback-style=none', '-style=file', '-i']
```
Comment (gerashegalov) [inline .pre-commit-config.yaml:10]:
```suggestion
                args: ['-fallback-style=none', '-style=file:thirdparty/cudf/cpp/.clang-format', '-i']
```

Comment (hyperbolic2346) [reply .pre-commit-config.yaml]: I was able to do this, but I had to update the clang-format version. In theory this doesn't matter, but I wanted to mention it.

**Review by gerashegalov — APPROVED** [review]

**Comment by hyperbolic2346** [comment]
build

**Comment by gerashegalov** [comment]
@hyperbolic2346  did you up-merge?

https://github.com/NVIDIA/spark-rapids-jni/runs/8214034914?check_suite_focus=true#step:2:21266
```
[2022-09-06T18:41:17.190Z] [INFO]      [exec]   CMake 3.23.1 or higher is required.  You are running version 3.22.3
```

**Comment by hyperbolic2346** [comment]
Thank you @gerashegalov. I had not.

**Comment by hyperbolic2346** [comment]
build

**Comment by jlowe** [comment]
An upmerge should not be necessary.  IIUC premerge runs on the result of merging the proposed PR changes into the dev branch, not on the dev branch directly.

**Comment by gerashegalov** [comment]
> An upmerge should not be necessary. IIUC premerge runs on the result of merging the proposed PR changes into the dev branch, not on the dev branch directly.

Are we getting a stale Docker image on CI then? /cc @pxLi

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
> > An upmerge should not be necessary. IIUC premerge runs on the result of merging the proposed PR changes into the dev branch, not on the dev branch directly.
> 
> Are we getting a stale Docker image on CI then? /cc @pxLi

We have updated jni image but looks like it has been overwritten by some elder source yesterday.
Now it should be on the correct image to run pre-merge, I will try figure out if any one upload an elder one by accident

**Pull request merged**